DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestForControllers/AccountControllerAssignRolesToUserTests.cs

```diff
@@ -0,0 +1,165 @@
+                    new List<SecurityRole>
+                    {
+                        new SecurityRole { Name = "Admin"}
+                    },
+                    new User
+                    {
+                        ProviderId = 1
+                    },
+                    1, 
+                    0, 
+                    "Admin" 
+                };
+            }
+        }
+
+        private ApplicationUserManager MockForUsers(List<User> users)
+        {
+            var userManager = Substitute.For<ApplicationUserManager>(Substitute.For<IUserStore<User>>());
+            //userManager.Users.Returns(new MockForEnumerableQuery<User>(users));
+            userManager.Users.Returns(users.AsQueryable());
+            return userManager;
+        }
+
+        private ISecurityContext MockForContext(IEnumerable<SecurityRole> roles)
+        {
+            var context =Substitute.For<ISecurityContext>();
+            context.SecurityRoles.Returns(new MockForDbSet<SecurityRole>(roles));
+            return context;
+        } 
+
+        private ApplicationUserManager MockForAddingClaim(List<User> users, User userToUpdate)
+        {
+            var userManager = MockForUsers(users);
+            userManager.UpdateAsync(userToUpdate).Returns(IdentityResult.Success);
+            userManager.CreateIdentityAsync(userToUpdate, DefaultAuthenticationTypes.ApplicationCookie).Returns(Task.FromResult(new ClaimsIdentity()));
+            userManager.AddClaimAsync(userToUpdate.Id, Arg.Any<Claim>()).Returns(IdentityResult.Success);
+            return userManager;
+        }
+
+        [Theory]
+        [MemberData("GetDataForNotFountResult")]
+        public void NotFoundUserTest(List<User> users, int gitHubId, int repoId, string roleToAssign)
+        {
+            AccountController controller = new AccountController(Substitute.For<IGithubService>(),
+                Substitute.For<ISecurityContext>(), MockForUsers(users));
+
+            Task<IHttpActionResult> response = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign);
+
+            IHttpActionResult result = response.Result;
+            result.Should().BeOfType<NotFoundResult>("Because user with providerId= {0} doesn't exists in database",gitHubId);
+        }
+
+        [Theory]
+        [MemberData("GetDataForInvalidModelStateResult")]
+        public void InvalidRoleTest(List<User> users, IEnumerable<SecurityRole> roles, int gitHubId, int repoId, string roleToAssign)
+        {
+            AccountController controller = new AccountController(Substitute.For<IGithubService>(),
+                MockForContext(roles), MockForUsers(users)); 
+
+            Task<IHttpActionResult> response = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign);
+
+            IHttpActionResult result = response.Result;
+            result.Should().BeOfType<InvalidModelStateResult>("Because impossible to assign role = {0} that doesn't exist in database", roleToAssign);
```

Comment:
It's just a message for developer if test will fail. For example,
"Expected <InvalidModelStateResult> Because impossible to assign role = {0} that doesn't exist in database<NotFoundResult>." ]
. This message are unic for every test. Do I still need to make a const for this message.??